src/connectors/index.ts

```diff
@@ -1,49 +1,64 @@
 import { Web3Provider } from '@ethersproject/providers'
-import { SafeAppConnector } from '@gnosis.pm/safe-apps-web3-react'
+// import { SafeAppConnector } from '@gnosis.pm/safe-apps-web3-react'
+import { CoinbaseWallet } from '@web3-react/coinbase-wallet'
+import { initializeConnector, Web3ReactHooks } from '@web3-react/core'
+import { MetaMask } from '@web3-react/metamask'
+import { Network } from '@web3-react/network'
+import { WalletConnect } from '@web3-react/walletconnect'
 import { ALL_SUPPORTED_CHAIN_IDS, SupportedChainId } from 'constants/chains'
 import { INFURA_NETWORK_URLS } from 'constants/infura'
-import { InjectedConnector } from 'web3-react-injected-connector'
-import { WalletConnectConnector } from 'web3-react-walletconnect-connector'
-import { WalletLinkConnector } from 'web3-react-walletlink-connector'
 
 import UNISWAP_LOGO_URL from '../assets/svg/logo.svg'
 import getLibrary from '../utils/getLibrary'
-import { FortmaticConnector } from './Fortmatic'
-import { NetworkConnector } from './NetworkConnector'
+// import { FortmaticConnector } from './Fortmatic'
 
-const FORMATIC_KEY = process.env.REACT_APP_FORTMATIC_KEY
+// const FORMATIC_KEY = process.env.REACT_APP_FORTMATIC_KEY
 
-export const network = new NetworkConnector({
-  urls: INFURA_NETWORK_URLS,
-  defaultChainId: 1,
-})
+export const [network, networkHooks] = initializeConnector<Network>(
+  (actions) => new Network(actions, INFURA_NETWORK_URLS, false, 1),
+  Object.keys(INFURA_NETWORK_URLS).map((chainId) => Number(chainId))
+)
 
 let networkLibrary: Web3Provider | undefined
 export function getNetworkLibrary(): Web3Provider {
   return (networkLibrary = networkLibrary ?? getLibrary(network.provider))
 }
 
-export const injected = new InjectedConnector({
-  supportedChainIds: ALL_SUPPORTED_CHAIN_IDS,
-})
-
-export const gnosisSafe = new SafeAppConnector()
-
-export const walletconnect = new WalletConnectConnector({
-  supportedChainIds: ALL_SUPPORTED_CHAIN_IDS,
-  rpc: INFURA_NETWORK_URLS,
-  qrcode: true,
-})
-
-// mainnet only
-export const fortmatic = new FortmaticConnector({
-  apiKey: FORMATIC_KEY ?? '',
-  chainId: 1,
-})
-
-export const walletlink = new WalletLinkConnector({
-  url: INFURA_NETWORK_URLS[SupportedChainId.MAINNET],
-  appName: 'Uniswap',
-  appLogoUrl: UNISWAP_LOGO_URL,
-  supportedChainIds: ALL_SUPPORTED_CHAIN_IDS,
-})
+export const [injected, injectedHooks] = initializeConnector<MetaMask>(
+  (actions) => new MetaMask(actions, { supportedChainIds: ALL_SUPPORTED_CHAIN_IDS })
```

Comment:
hm this doesn't look quite right - 1st argument to `initializeConnector` is the (actions) => function, but the MetaMask connector constructor takes the following args: https://github.com/NoahZinsmeister/web3-react/blob/3e0a0785f27a1a0413a2f81d2b0669c017bd4bf0/packages/metamask/src/index.ts#L36

also, the supported chain ids argument should be passed as the 2nd arg, see https://github.com/NoahZinsmeister/web3-react/blob/3e0a0785f27a1a0413a2f81d2b0669c017bd4bf0/packages/core/src/hooks.ts#L28